### ISSUE TITLE
Cannot create 0-element Block domains or arrays

### ISSUE THREAD

**Issue by mppf** [chapel-lang/chapel#12857]
``` chapel
use BlockDist;
var A = newBlockArr(1..0, int);
```

results in a runtime error
```
error: halt reached - Block() requires a non-empty boundingBox
```

This pattern would be useful in the event re-distributing a Block array is supported (e.g. as in  #5938). But even if that is not supported, it seems like a corner case that should work.

**Comment by bradcray** [comment]
I agree and could take the action to fix this if nobody else is motivated to do this.  I've had a mental note to tackle this since learning about it, but hadn't gotten around to it yet.
